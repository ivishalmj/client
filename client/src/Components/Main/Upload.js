/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/no-access-state-in-setstate */
import React, { Component } from 'react';
import Uppy from '@uppy/core';
import Tus from '@uppy/tus';
import GoogleDrive from '@uppy/google-drive';
import XHRUpload from '@uppy/xhr-upload';
import DashboardModal from '@uppy/react/lib/DashboardModal';
import { Button } from 'semantic-ui-react';
import '@uppy/core/dist/style.css';
import '@uppy/dashboard/dist/style.css';

export default class DragAndDropFileUploader extends Component {
  constructor(props) {
    super(props);

    this.state = {
      open: false,
    };
    this.handleModalClick = this.handleModalClick.bind(this);

    this.uppy = Uppy({ id: 'uppy1', autoProceed: true, debug: true })
      .use(Tus, { endpoint: '/api/v1/documents' })
      .use(GoogleDrive, { companionUrl: 'https://companion.uppy.io' })
      .use(XHRUpload, {
        endpoint: '/api/v1/policies',
        formData: true,
        fieldName: 'files[]',
      });
    // console.log(this.uppy);
  }

  componentWillUnmount() {
    this.uppy.close();
  }

  handleModalClick() {
    this.setState({
      open: !this.state.open,
    });
  }

  render() {
    const { open } = this.state;

    return (
      // eslint-disable-next-line react/jsx-filename-extension
      <div>
        <div>
          <Button id="dashboardmodal" onClick={this.handleModalClick}>
            {open ? 'Close dashboard' : 'Upload Documents'}
          </Button>
          <DashboardModal
            uppy={this.uppy}
            open={open}
            target={document.body}
            onRequestClose={() => this.setState({ open: false })}
            plugins={['Webcam']}
            trigger="#dashboardmodal"
          />
        </div>
      </div>
    );
  }
}
