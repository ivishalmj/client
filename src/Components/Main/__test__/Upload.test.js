
/**
 * @jest-environment jsdom
 */
import React from 'react';
import ReactDOM from "react-dom";
import {render, fireEvent,} from '@testing-library/react';
import Upload from '../Upload';

it('<Upload/> Matches Snapshot', () => {
  const component = render(<Upload/>);
  expect(component).toMatchSnapshot();
});

it("<Upload/> Renders without Crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<Upload/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("Inital State open: false", () => {
  const { getByText}  = render( <Upload/>);
  const button = getByText('Upload Documents');

  fireEvent.click(button)
  expect((button).textContent).toBe('Close dashboard')

});

