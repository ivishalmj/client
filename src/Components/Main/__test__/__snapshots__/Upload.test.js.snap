// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Upload/> Matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <button
            class="ui button"
            id="dashboardmodal"
          >
            Upload Documents
          </button>
          <div />
        </div>
      </div>
    </div>
    <div
      class="uppy-Root"
      dir="ltr"
    >
      <div
        aria-hidden="true"
        aria-label="File Uploader Window (Press escape to close)"
        class="uppy-Dashboard uppy-Dashboard--animateOpenClose uppy-Dashboard--modal"
        data-uppy-drag-drop-supported="true"
        data-uppy-num-acquirers="0"
        data-uppy-theme="light"
      >
        <div
          class="uppy-Dashboard-overlay"
          tabindex="-1"
        />
        <div
          aria-modal="true"
          class="uppy-Dashboard-inner"
          role="dialog"
        >
          <button
            aria-label="Close Modal"
            class="uppy-u-reset uppy-Dashboard-close"
            title="Close Modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
          <div
            class="uppy-Dashboard-innerWrap"
          >
            <div
              class="uppy-Dashboard-dropFilesHereHint"
            >
              Drop your files here
            </div>
            
            <div
              class="uppy-Dashboard-AddFiles"
            >
              <input
                aria-hidden="true"
                class="uppy-Dashboard-input"
                hidden=""
                multiple=""
                name="files[]"
                tabindex="-1"
                type="file"
              />
              <input
                aria-hidden="true"
                class="uppy-Dashboard-input"
                hidden=""
                multiple=""
                name="files[]"
                tabindex="-1"
                type="file"
                webkitdirectory="true"
              />
              <div
                class="uppy-Dashboard-AddFiles-title"
              >
                Drop files here, paste or 
                <button
                  class="uppy-u-reset uppy-Dashboard-browse"
                  data-uppy-super-focusable="true"
                  type="button"
                >
                  browse files
                </button>
              </div>
              
              <div
                class="uppy-Dashboard-AddFiles-info"
              >
                
                <a
                  class="uppy-Dashboard-poweredBy"
                  href="https://uppy.io"
                  rel="noreferrer noopener"
                  tabindex="-1"
                  target="_blank"
                >
                  Powered by 
                  <span>
                    <svg
                      aria-hidden="true"
                      class="uppy-c-icon uppy-Dashboard-poweredByIcon"
                      focusable="false"
                      height="11"
                      viewBox="0 0 11 11"
                      width="11"
                    >
                      <path
                        d="M7.365 10.5l-.01-4.045h2.612L5.5.806l-4.467 5.65h2.604l.01 4.044h3.718z"
                        fill-rule="evenodd"
                      />
                    </svg>
                    <span
                      class="uppy-Dashboard-poweredByUppy"
                    >
                      Uppy
                    </span>
                  </span>
                </a>
              </div>
            </div>
            
            
            
            
            <div
              class="uppy-Dashboard-progressindicators"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <button
          class="ui button"
          id="dashboardmodal"
        >
          Upload Documents
        </button>
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
